repo_state_checks/check_repo_structure.py

```diff
@@ -0,0 +1,56 @@
+"""
+Checks to see if repo structure
+"""
+
+import glob
+import os
+import re
+import pdb
+
+module_dict_key = 'repo_structure'
+
+def check_python_files(all_results, repo_path):
+    """
+    See if current repo has python files in it
+    Only checks one level down(repo_path/*/*.py)
+    """
+    # first check if py exist in head file
+    python_files = glob.glob(repo_path + "/*.py", recursive=True)
+    all_results[module_dict_key]['has_python_files'] = False
+    if len(python_files) > 0:
+        all_results[module_dict_key]['has_python_files'] = True
+    # if no py files in head dir, check below
+    if not all_results[module_dict_key]['has_python_files']:
+        subdirs = glob.glob('*')
+        for name in subdirs:
+            if os.path.isdir(repo_path + '/'+ name):
+                python_files = glob.glob(repo_path + '/'+ name + "/*.py", recursive=True)
+                if len(python_files) > 0:
+                    all_results[module_dict_key]['has_python_files'] = True
+
+
+
+def check_necessary_files(all_results, repo_path):
+    """ Test if each repo has required files """
+    necessary_files = ['openedx.yaml', 'tox.ini', 'Makefile', '.travis.yml', 'README.rst', 'pylintrc']
```

Comment:
Given that we have different kinds of repositories (Python, JS, etc.), we may want to keep the file existence checks grouped with other checks of the content of each file.